compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -619,20 +603,17 @@ impl<'tcx> InferCtxtBuilder<'tcx> {
         })
     }
 
-    pub fn enter<R>(&mut self, f: impl for<'a> FnOnce(InferCtxt<'a, 'tcx>) -> R) -> R {
+    pub fn enter<R>(&mut self, f: impl for<'a> FnOnce(InferCtxt<'tcx>) -> R) -> R {
```

Comment:
same here: `'a` is unused :thinking: I think we should either just return an `InferCtxt` here or actually try to change this to `&'a InferCtxt<'tcx>`. Using a closure has the advantage that `enter` can check that `R` does not contain any inference variables afterwards.

Considering that we don't currently check this in `enter`, it seems sensible to just return an `InferCtxt` here.

Reply:
cc @rust-lang/types about changing `enter` here

Reply:
Should it be named `build` then? And `build_with_canonical`?

Reply:
jup :+1: renaming this to enter makes sense to me

Reply:
I'll assume you meant to say `build` 😄